src/components/password/password.component.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useRef, useState } from "react";
+import { TextboxProps } from "../textbox";
+import { StyledPassword, HiddenAriaLive } from "./password.style";
+import guid from "../../__internal__/utils/helpers/guid";
+import useLocale from "../../hooks/__internal__/useLocale";
+import ButtonMinor from "../button-minor/button-minor.component";
+
+export type TypeOptions = "text" | "password";
+
+export interface PasswordProps extends TextboxProps {
+  /** HTML type attribute of the input */
+  type?: TypeOptions;
+}
+
+export const Password = ({ id, type, ...rest }: PasswordProps) => {
```

Comment:
`type` is declared here but I don't think you've used it anywhere.

Reply:
Used on `StyledPassword`, which is a styled `Textbox` 👍

Reply:
Ah nvm I see what you mean, i'm looking at moving towards `type` being used in a hook, opposed to boolean which we use at the minute. That way we can still change the input type and icons based on the current state, but also allow users to change `type` manually, for use cases like changing back to `password` form submission etc